tests/unit/amo/components/TestSearchForm.js

```diff
@@ -219,6 +219,173 @@ describe(__filename, () => {
       }));
     });
 
+    it('does not fetch suggestions twice on focus when search has not changed', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      const dispatch = sinon.spy();
+      const wrapper = mountComponent({
+        dispatch,
+        suggestions: autocompleteState.suggestions,
+      });
+
+      sinon.assert.notCalled(dispatch);
+
+      // User types 'a' in the search input.
+      wrapper.find('input').simulate('change', createFakeChangeEvent('a'));
+
+      // User clicks somewhere else on the UI, triggering the
```

Comment:
These comments in the test describing the actions the user would take are fantastic, thank you much!